File: frontend/travel-with-anna-ui/src/app/services/models/budget-expenses-respond.ts
/* tslint:disable */
/* eslint-disable */
import { BudgetResponse } from '../models/budget-response';
import { ExpanseByCurrency } from '../models/expanse-by-currency';
import { ExpanseResponse } from '../models/expanse-response';
export interface BudgetExpensesRespond {
  budgetResponse?: BudgetResponse;
  expanses?: Array<ExpanseResponse>;
  overallPaidInTripCurrency?: number;
  overallPriceInTripCurrency?: number;
  paidBalanceDue?: number;
  priceBalanceDue?: number;
  sumsByCurrency?: Array<ExpanseByCurrency>;
  totalDebtInTripCurrency?: number;
}

File: backend/travel-with-anna/src/main/java/com/andwis/travel_with_anna/trip/budget/BudgetExpensesRespond.java
package com.andwis.travel_with_anna.trip.budget;

import com.andwis.travel_with_anna.trip.expanse.ExpanseByCurrency;
import com.andwis.travel_with_anna.trip.expanse.ExpanseResponse;

import java.math.BigDecimal;
import java.util.List;

public record BudgetExpensesRespond(
        BudgetResponse budgetResponse,
        List<ExpanseResponse> expanses,
        List<ExpanseByCurrency> sumsByCurrency,
        BigDecimal overallPriceInTripCurrency,
        BigDecimal overallPaidInTripCurrency,
        BigDecimal totalDebtInTripCurrency,
        BigDecimal priceBalanceDue,
        BigDecimal paidBalanceDue
) {
}
